### PR TITLE
add run energy regen meter

### PR DESCRIPTION
**Pull request by raiyni**
It's not always accurate but added a case to sync it back up.

closes https://github.com/runelite/runelite/issues/5166

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java [hunk 1]
```diff
@@ -41,17 +42,20 @@
 import net.runelite.client.config.ConfigManager;
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.runenergy.RunEnergyPlugin;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: You should definitely not reference other plugins like this

**Comment by ShaunDreclin** [comment]
Wouldn't this be better merged into the run energy plugin?

**Comment by raiyni** [comment]
Considering it's very specific to the regeneration meters, no.

**Comment by ShaunDreclin** [comment]
Oh I must have misread something, didnt notice it was part of a bigger regen plugin

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java (hunk 1 above)
Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: changed to be like other utils

**Comment by raiyni** [comment]
Not modding the tick value makes it more consistent with the regen.

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import static net.runelite.api.ItemID.*;
+
+@Singleton
+public class Graceful
+{
+	@Inject
+	private Client client;
+
+	// TODO It would be nice if we have the IDs for just the equipped variants of the Graceful set items.
+	private static final ImmutableSet<Integer> ALL_GRACEFUL_HOODS = ImmutableSet.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: I would rather just make this entire class enum

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	CAPE(
+		GRACEFUL_CAPE_11853, GRACEFUL_CAPE_13581, GRACEFUL_CAPE_13582, GRACEFUL_CAPE_13593, GRACEFUL_CAPE_13594,
+		GRACEFUL_CAPE_13605, GRACEFUL_CAPE_13606, GRACEFUL_CAPE_13617, GRACEFUL_CAPE_13618, GRACEFUL_CAPE_13629,
+		GRACEFUL_CAPE_13630, GRACEFUL_CAPE_13669, GRACEFUL_CAPE_13670, GRACEFUL_CAPE_21064, GRACEFUL_CAPE_21066,
+		AGILITY_CAPE, AGILITY_CAPET, MAX_CAPE
+	);
+
+	private int[] ids;
+
+	private static final Set<Integer> itemIds = new HashSet<>();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: This seems unnecessary? Just use respective enum's id array for each body slot

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
+package net.runelite.client.util;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: License header

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Set;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import static net.runelite.api.ItemID.*;
+
+public enum Graceful
+{
+
+	// TODO It would be nice if we have the IDs for just the equipped variants of the Graceful set items.
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: TODO:, not just TODO

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.HashSet;
+import java.util.Set;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import static net.runelite.api.ItemID.*;
+
+public enum Graceful
+{
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: Remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java [hunk 7]
```diff
@@ -45,54 +41,17 @@
...
 import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.util.Graceful;
 import org.apache.commons.lang3.StringUtils;
 
 @PluginDescriptor(
 	name = "Run Energy",
 	description = "Show various information related to run energy",
 	tags = {"overlay", "stamina"}
 )
+@Slf4j
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java]: I dont see logger being used

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java [hunk 8]
```diff
@@ -150,4 +173,14 @@ else if (currentHP > maxHP)
 			hitpointsPercentage = 1 - hitpointsPercentage;
 		}
 	}
+
+	private double run_regen_per_tick()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: Naming? camelCase

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterConfig.java [hunk 9]
```diff
@@ -34,7 +34,8 @@
 	@ConfigItem(
 		keyName = "showHitpoints",
 		name = "Show hitpoints regen",
-		description = "Show a ring around the hitpoints orb")
+		description = "Show a ring around the hitpoints orb",
+		position = 1)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterConfig.java]: we usually put ) on new lines

File: runelite-client/src/main/java/net/runelite/client/util/Graceful.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+		GRACEFUL_HOOD_13628, GRACEFUL_HOOD_13667, GRACEFUL_HOOD_13668, GRACEFUL_HOOD_21061, GRACEFUL_HOOD_21063
+	)),
+
+	TOP(ImmutableSet.of(
+		GRACEFUL_TOP_11855, GRACEFUL_TOP_13583, GRACEFUL_TOP_13584, GRACEFUL_TOP_13595, GRACEFUL_TOP_13596,
+		GRACEFUL_TOP_13607, GRACEFUL_TOP_13608, GRACEFUL_TOP_13619, GRACEFUL_TOP_13620, GRACEFUL_TOP_13631,
+		GRACEFUL_TOP_13632, GRACEFUL_TOP_13671, GRACEFUL_TOP_13672, GRACEFUL_TOP_21067, GRACEFUL_TOP_21069
+	)),
+
+	LEGS(ImmutableSet.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/Graceful.java]: You can stiull use varargs ctor, and just create the immutable set from it to reduce boilerplate

**Comment by raiyni** [comment]
updated to not tick while running

**Comment by raiyni** [comment]
I know it was discussed in discord, the usefulness of having the Graceful being a util. I don't disagree that it probably isn't useful anywhere else.

Status orb plugins should be combined into 1 plugin then. The Graceful can then live in there.

**Pull request closed without merging**